stdlib/join/merge_join.go

```diff
@@ -0,0 +1,522 @@
+package join
+
+import (
+	"context"
+	"sync"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+// MergeJoinTransformation performs a sort-merge-join on two table streams.
+// It makes the following assumptions about the input tables:
+//   * They have the same group key
```

Comment:
I guess you mean that they have the same group key columns?

Reply:
Yeah, I now realize that this was poorly worded. I'll amend it.